processor/src/advice/providers.rs

```diff
@@ -307,6 +307,20 @@ impl AdviceProvider for MemAdviceProvider {
     }
 }
 
+impl MemAdviceProvider {
+    // FINALIZATION
+    // --------------------------------------------------------------------------------------------
+    pub fn into_parts(self) -> (u32, Vec<Felt>, SimpleAdviceMap, MerkleStore) {
+        let BaseAdviceProvider {
+            step,
+            stack,
+            map,
+            store,
+        } = self.provider;
+        (step, stack, map, store)
+    }
```

Comment:
I wonder if we should include `step` in this. It is used solely for error reporting purposes and ideally should be removed from advice providers at some point.